core/src/main/java/org/apache/brooklyn/core/config/BasicConfigKey.java

```diff
@@ -331,12 +337,19 @@ public ConfigInheritance getParentInheritance() {
         return parentInheritance;
     }
 
+    /** @see ConfigKey#getConstraints() */
+    @Nonnull
+    @Override
+    public Map<String, Predicate<? super T>> getConstraints() {
+        return constraints;
+    }
+
     /** @see ConfigKey#getConstraint() */
     @Override @Nonnull
     public Predicate<? super T> getConstraint() {
         // Could be null after rebinding
-        if (constraint != null) {
-            return constraint;
+        if (constraints.size() > 0) {
+            return Predicates.and(constraints.values());
         } else {
             return Predicates.alwaysTrue();
         }
```

Comment:
What we care about is the "required" as implemented in yaml (`isNonBlank`). It's enought to support that, but could easily add some more to a white-list. Also alias to it as a "required" predicate to be used in Java if anyone wants to integrate with UI better from Java land.

Also keep in mind backwards compatibility (always :) ). Instead of changing the API straight away we'd have to go through a deprecation cycle where the old signatures are still supported.  That's one more reason why stick to the existing non-labeled predicates.